botorch/posteriors/ensemble.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+r"""
+Ensemble posteriors. Used in conjunction with ensemble models.
+"""
+
+from __future__ import annotations
+
+from typing import Optional
+
+import torch
+from botorch.posteriors.posterior import Posterior
+from torch import Tensor
+
+r"""
+Posterior module to be used with ensemble models.
+"""
+
+
+class EnsemblePosterior(Posterior):
+    r"""Ensemble posterior, that should be used for ensemble models that compute
+    eagerly a finite number of samples per X value as for example a deep ensemble
+    or a random forest."""
+
+    def __init__(self, values: Tensor) -> None:
+        r"""
+        Args:
+            values: Values of the samples produced by this posterior as
+                a `(b) x s x q x m` tensor where `m` is the output size of the
+                model and `s` is the ensemble size.
+        """
+        if values.ndim < 3:
+            raise ValueError("Values has to be at least three-dimensional.")
+        self.values = values
+
+    @property
+    def size(self) -> int:
```

Comment:
let's call this sth like `ensemble_size` to be unambiguous; `size()` is typically a alias for `shape`

Reply:
Done.